geo/parse.go

```diff
@@ -272,29 +275,38 @@ func extract4DCoordinates(thing interface{}) (rv [][][][]float64) {
 	return rv
 }
 
-func extractGeoShape(thing interface{}) ([][][][]float64, string, bool) {
+func ParseGeoShapeField(thing interface{}) (interface{}, string, error) {
 	thingVal := reflect.ValueOf(thing)
 	if !thingVal.IsValid() {
-		return nil, "", false
+		return nil, "", nil
 	}
 
+	var shape string
 	var coordValue interface{}
-	var typ string
+
 	if thingVal.Kind() == reflect.Map {
 		iter := thingVal.MapRange()
 		for iter.Next() {
 			if iter.Key().String() == "type" {
-				typ = iter.Value().Interface().(string)
-				// make the shape type case insensitive.
-				typ = strings.ToLower(typ)
```

Comment:
Would we be giving up on the case insensitivity here ?

Reply:
Nope, it is handled in the new function in parse.go line no:301.

Reply:
Ack.